### ISSUE TITLE
Wrong Lint/UnneededSplatExpansion inside `case` clause

### ISSUE THREAD

**Issue by samphilipd** [bbatsov/rubocop#4269]
Erroneous `Lint/UnneededSplatExpansion` when matching percentage delimited string literals in a `case` statement.

## Example:

```ruby
x = 'a'
case x
when *%w[a b c] # <= Rubocop erroneously marks the splat here as unnecessary.
  puts "It matched"
else
  puts "It didn't match"
end
```

```
$ rubocop -V
0.48.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
```

**Comment by rrosenblum** [comment]
RuboCop appears to be behaving properly here to me. The splat expansion is unnecessary because it is being used to expand an array whose values can be represented as a list of arguments.  

Running RuboCop with auto-correct produces
```ruby
x = 'a' 
case x
when 'a', 'b', 'c' # <= This is correct
  puts "It matched"
else
  puts "It didn't match"
end
```

The list of arguments is preferred because `*%w[a b c]` causes Ruby to have to evaluate the contents of the array and expand it every time that this condition is called.

@samphilipd can you provide more information as to why this might not be working properly?

**Comment by samphilipd** [comment]
@rrosenblum Ah, I see that you can expand the splat. Well, I suppose that's ok it's only a few extra characters. This issue can be closed.

**Comment by rrosenblum** [comment]
> it's only a few extra characters

@samphilipd This has little to do with the number of characters, and a lot to do with how Ruby handles splat expansions. 

```ruby
[7] pry(main)> def slow
[7] pry(main)*   x = 'a'
[7] pry(main)*   case x
[7] pry(main)*   when *%w[a b c]  
[7] pry(main)*     nil
[7] pry(main)*   else  
[7] pry(main)*     nil
[7] pry(main)*   end  
[7] pry(main)* end  
=> :slow
[8] pry(main)> def fast
[8] pry(main)*   x = 'a'
[8] pry(main)*   case x
[8] pry(main)*   when 'a', 'b', 'c'  
[8] pry(main)*     nil
[8] pry(main)*   else  
[8] pry(main)*     nil
[8] pry(main)*   end  
[8] pry(main)* end  
=> :fast
[9] pry(main)> Benchmark.ips do |x|
[9] pry(main)*   x.report('splat') { slow }
[9] pry(main)*   x.report('no splat') { fast }
[9] pry(main)*   x.compare!
[9] pry(main)* end  
Warming up --------------------------------------
               splat   188.764k i/100ms
            no splat   255.615k i/100ms
Calculating -------------------------------------
               splat      3.222M (± 3.2%) i/s -     16.234M in   5.044293s
            no splat      6.042M (± 3.4%) i/s -     30.418M in   5.040205s

Comparison:
            no splat:  6042309.4 i/s
               splat:  3221539.2 i/s - 1.88x  slower
```

As the number elements in the array grows, the performance gets worse. It is unlikely that this will be the bottleneck in any system, however having something that automatically helps the situation is nice. This check used to be handled by `Performance/CaseWhenSplat`, but this was moved out to the more generic `Lint/UnneededSplatExpansion` cop.

**Comment by JelF** [comment]
I sure this explanation should be referenced in cop description. Right now this is
`Description: Checks for splat unnecessarily being called on literals`, which is irrelevant
